include/mbedtls/pkcs7.h

```diff
@@ -0,0 +1,238 @@
+/**
+ * \file pkcs7.h
+ *
+ * \brief PKCS7 generic defines and structures
+ *  https://tools.ietf.org/html/rfc2315
+ */
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * Note: For the time being, this application of the PKCS7 cryptographic
+ * message syntax is a partial implementation of RFC 2315.
+ * Differences include:
+ *  - The RFC specifies 6 different content types. The only type currently
+ *    supported in MbedTLS is the signed data content type.
+ *  - The only supported PKCS7 Signed Data syntax version is version 1
+ *  - The RFC specifies support for BER. This application is limited to
+ *    DER only.
+ *  - The RFC specifies that multiple digest algorithms can be specified
+ *    in the Signed Data type. Only one digest algorithm is supported in MbedTLS.
+ *  - The RFC specifies the Signed Data certificate format can be
+ *    X509 or PKCS6. The only type currently supported in MbedTLS is X509.
+ *  - The RFC specifies the Signed Data type can contain
+ *    certificate-revocation lists (crls). This application has no support
+ *    for crls so it is assumed to be an empty list.
+ *  - The RFC specifies support for multiple signers. This application only
```

Comment:
It looks like the latest version of this PR adds support for multiple signers?

Reply:
Whoops, yes. I think I was trying to explain that only one certificate can be listed in the `certificates` field of the `SignedData` type. I will fix.